### ISSUE TITLE
detect invalid nodes and invalid graph in onnxruntime before handing them to execution providers

### ISSUE THREAD

**Issue by jeyblu** [microsoft/onnxruntime#8393]
### Discussed in https://github.com/microsoft/onnxruntime/discussions/8152

<div type='discussions-op-text'>

<sup>Originally posted by **jeyblu** June 24, 2021</sup>
Tests like [TEST(BatchNormTest, InvalidScaleDim)](https://github.com/microsoft/onnxruntime/blob/378a98597e2c61ccd11343906bd8d83d1af6b7c7/onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc#L406) have invalid values in the node attribute to test if invalid nodes are detected. It seems that it would make more sense for onnxruntime to check nodes in the graph and detect invalid ones before handing the graph to execution providers. If it's checked by onnxruntime, the code is written once in onnxruntime. If it's checked by execution providers, the code has to be written by every execution provider. It's one implementation vs n number of implementations for n number of execution providers.</div>
